### PR TITLE
HAWQ-1591 Common tuple batch structure for VecExe

### PR DESCRIPTION
**Pull request by weinan003**
Define *TupleBatchData* struct to hold a batch of tuples through vectorized execution process.

### CODE REVIEW

File: contrib/vexecutor/tuplebatch.c [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    //deserial member value -- datagroup
+    if(tb->ncols != 0)
+    {
+        tb->datagroup = palloc0(sizeof(vheader*) * tb->ncols);
+        int colid;
+        while (len < buflen)
+        {
+            memcpy(&colid,buffer + len,sizeof(int));
+            len += sizeof(int);
+            tb->datagroup[colid] = (vheader* ) GetVFunc(((vheader *) (buffer + len))->elemtype)->deserialization(buffer + len,&tmplen);
```
Comment (wengyanqing) [inline contrib/vexecutor/tuplebatch.c]: Do the serialization and deserialization always succeed ?

File: contrib/vexecutor/tuplebatch.h [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    int     _tb_len;
+    int     batchsize;  //indicate maximum number of batch
+    int     ncols;      //the number of target table column
+    int     nrows;      //the number of tuples scaned out
+    int     iter;       //used for data
+    bool*   skip;       //used for qualification
+    vheader** datagroup;
+}TupleBatchData,*TupleBatch;
+
+/* TupleBatch ctor */
```
Comment (wengyanqing) [inline contrib/vexecutor/tuplebatch.h]: 1. Would you give a user case of tuplebatch struct or the function call logic of its functions ?
2. How the tuple batch is integrated with current TupleTableSlot ?

File: contrib/vexecutor/tuplebatch.h [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+#ifndef __TUPLEBATCH_H__
+#define __TUPLEBATCH_H__
+
+#include "vexecutor.h"
+#include "vcheck.h"
+
+
+typedef struct TupleBatchData
+{
+    int     _tb_len;
```
Comment (interma) [inline contrib/vexecutor/tuplebatch.h]: _tb_len no use?

Comment (weinan003) [reply contrib/vexecutor/tuplebatch.h]: pg some lib function will require an int type variable as the total size of the struct recorder

File: contrib/vexecutor/tuplebatch.c [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+        elog(FATAL,"TupleBatch Allocation failed");
+        return NULL;
+    }
+
+    tb->ncols = colnum;
+    tb->batchsize = batchsize;
+
+    tb->skip = palloc0(sizeof(bool) * tb->batchsize);
+    tb->datagroup = palloc0(sizeof(struct vtypeheader*) * tb->ncols);
+
```
Comment (wengyanqing) [inline contrib/vexecutor/tuplebatch.c]: It's better to init all the struct variable members (nrow, iter) in tbGenerate or call tbReset before return.

**Comment by wengyanqing** [comment]
LGTM

**Pull request closed without merging**